### PR TITLE
[Documentation] Document cart endpoints

### PR DESCRIPTION
**Pull request by lchrusciel**

### CODE REVIEW

File: doc/swagger.yml [hunk 1]
```diff
@@ -0,0 +1,1135 @@
...
+  externalDocs:
+    description: "Find out more"
+    url: "http://docs.sylius.org/en/latest/book/products/index.html"
+- name: "taxon"
+  description: "Show taxon tree"
+  externalDocs:
+    description: "Find out more"
+    url: "http://docs.sylius.org/en/latest/book/products/taxons.html"
+- name: "checkout"
+  description: "All actions related to checkout fulfillment. It is important , to execute them in given order (address, choose shipment, choose payment and complete)."
```
Comment (venyii) [inline doc/swagger.yml]: Removable space after `It  is important`

File: doc/swagger.yml [hunk 2]
```diff
@@ -0,0 +1,1135 @@
...
+    - name: "token"
+      in: "path"
+      description: "Cart identifier."
+      required: true
+      type: "string"
+    post:
+      tags:
+      - "cart"
+      summary: "Pick up your cart from the store"
+      description: "This endpoint will allow you to assign a new cart to provided token. We recommend using UUID as a token to avoid duplication. If any of previous cart or order will have same token value an exception will be thrown."
```
Comment (venyii) [inline doc/swagger.yml]: to **the** provided token

Comment (venyii) [inline doc/swagger.yml]: If any of the previous carts or orders already have the same token, ...

File: doc/swagger.yml [hunk 3]
```diff
@@ -0,0 +1,1135 @@
...
+      operationId: "cartPickUp"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "content"
+        description: "Contains an information about channel which should be associated with newly created cart"
```
Comment (venyii) [inline doc/swagger.yml]: about **the** channel
with **the** newly created...

File: doc/swagger.yml [hunk 4]
```diff
@@ -0,0 +1,1135 @@
...
+            $ref: "#/definitions/Cart"
+        400:
+          description: "Invalid input (E.g. token has not been found)"
+          schema:
+            $ref: "#/definitions/GeneralError"
+    delete:
+      tags:
+      - "cart"
+      summary: "Drop your cart."
+      description: "This endpoint will remove cart and all of related cart items."
```
Comment (venyii) [inline doc/swagger.yml]: remove **the** cart
all of **the** related...

File: doc/swagger.yml [hunk 5]
```diff
@@ -0,0 +1,1135 @@
...
+          description: "Cart has been picked up and assigned to given token"
+        400:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/GeneralError"
+    get:
+      tags:
+      - "cart"
+      summary: "Show summarized cart."
+      description: "This endpoint show you current calculated state of cart."
```
Comment (venyii) [inline doc/swagger.yml]: This endpoint show**s** you **the** current...

**Comment by lchrusciel** [comment]
@venyii any feedback is valuable! :) If you can iterate one more time and point me out all problems, I will be really grateful :)

File: doc/swagger.yml [hunk 6]
```diff
@@ -0,0 +1,1135 @@
...
+    url: "http://docs.sylius.org/en/latest/book/products/taxons.html"
+- name: "checkout"
+  description: "All actions related to checkout fulfillment. It is important , to execute them in given order (address, choose shipment, choose payment and complete)."
+  externalDocs:
+    description: "Find out more"
+    url: "http://docs.sylius.org/en/latest/book/orders/checkout.html"
+schemes:
+- "https"
+paths:
+  /cart/{token}:
```
Comment (lchrusciel) [inline doc/swagger.yml]: `cart` -> `carts`

File: doc/swagger.yml [hunk 7]
```diff
@@ -0,0 +1,1135 @@
...
+        400:
+          description: "Invalid input (E.g. token has not been found)"
+          schema:
+            $ref: "#/definitions/GeneralError"
+  /checkout/{token}/address:
+    put:
+      tags:
+      - "checkout"
+      summary: "Address cart."
+      description: "This endpoint will allow you to address cart and begin the checkout process. You can either define same shipping and billing addresses or send specify them separately."
```
Comment (venyii) [inline doc/swagger.yml]: This endpoint will allow you to add billing and shipping addresses to the cart and begin the checkout process. You can either define the same shipping and billing address or specify them separately.

File: doc/swagger.yml [hunk 8]
```diff
@@ -0,0 +1,1135 @@
...
+      - "application/json"
+      parameters:
+      - name: "token"
+        in: "path"
+        description: "Cart identifier."
+        required: true
+        type: "string"
+      responses:
+        200:
+          description: "Current state of cart, with calculated prices and related items."
```
Comment (venyii) [inline doc/swagger.yml]: of **the** cart

File: doc/swagger.yml [hunk 9]
```diff
@@ -0,0 +1,1135 @@
...
+        400:
+          description: "Invalid input, validation failed."
+          schema:
+            $ref: "#/definitions/GeneralError"
+  /checkout/{token}:
+    get:
+      tags:
+      - "checkout"
+      summary: "Show checkout summarizon"
+      description: "This endpoint will show summarized cart during checkout. This action is an equivalent of cart summarize action."
```
Comment (venyii) [inline doc/swagger.yml]: show **the** summarized

File: doc/swagger.yml [hunk 10]
```diff
@@ -0,0 +1,1135 @@
...
+      summary: "Show summarized cart."
+      description: "This endpoint show you current calculated state of cart."
+      operationId: "cartSummarize"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "Current state of cart, with calculated prices and related items."
```
Comment (venyii) [inline doc/swagger.yml]: of **the** cart

File: doc/swagger.yml [hunk 11]
```diff
@@ -0,0 +1,1135 @@
...
+      type: "string"
+    - name: "identifier"
+      in: "path"
+      description: "Identifier of specific item. Can be found on cart summarizon."
+      required: true
+      type: "string"
+    put:
+      tags:
+      - "cart"
+      summary: "Change quantity of cart item."
```
Comment (venyii) [inline doc/swagger.yml]: of **a** cart item

File: doc/swagger.yml [hunk 12]
```diff
@@ -0,0 +1,1135 @@
...
+  /cart/{token}/items/{identifier}:
+    parameters:
+    - name: "token"
+      in: "path"
+      description: "Cart identifier."
+      required: true
+      type: "string"
+    - name: "identifier"
+      in: "path"
+      description: "Identifier of specific item. Can be found on cart summarizon."
```
Comment (venyii) [inline doc/swagger.yml]: Identifier of **a** specific item. Can be found in **the** cart **summary**.

File: doc/swagger.yml [hunk 13]
```diff
@@ -0,0 +1,1135 @@
...
+          description: "Coupon has been assigned to the cart."
+        400:
+          description: "Invalid input, validation failed."
+          schema:
+            $ref: "#/definitions/GeneralError"
+  /checkout/{token}:
+    get:
+      tags:
+      - "checkout"
+      summary: "Show checkout summarizon"
```
Comment (venyii) [inline doc/swagger.yml]: Show checkout **summary**

File: doc/swagger.yml [hunk 14]
```diff
@@ -0,0 +1,1135 @@
...
+      description: "This endpoint will allow you to put a new item to your cart."
+      operationId: "cartPutItem"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "content"
+        description: "Description of an item. The smallest required amount of data is a product code and quantity for simple product. Configurable products will require additional `variant_code` or `options` but never both."
```
Comment (venyii) [inline doc/swagger.yml]: ...for **a** simple product
...will require **an** additional `variant_code` or `options` **field**, but never both.

File: doc/swagger.yml [hunk 15]
```diff
@@ -0,0 +1,1135 @@
...
+    - name: "token"
+      in: "path"
+      description: "Cart identifier."
+      required: true
+      type: "string"
+    post:
+      tags:
+      - "cart"
+      summary: "Add an item to your cart."
+      description: "This endpoint will allow you to put a new item to your cart."
```
Comment (venyii) [inline doc/swagger.yml]: Since you were using the term `add` before, I'd use it here too

...allow you to **add** a new...

**Comment by pjedrzejewski** [comment]
@lchrusciel Looks good to me, please updated it after reviews and it is good to merge. Great work Łukasz, thanks!

**Comment by lchrusciel** [comment]
Thanks, @venyii! I owe you a 🍺  when we meet :D

**Pull request merged**